deployment/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,392 @@
+package elasticsearch
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	es "github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
```

Comment:
@agarciamontoro - Note that we are still using the v7 ES client here rather than the official opesearch client: https://github.com/opensearch-project/opensearch-go. Perhaps this should be added as a follow up ticket to use the opensearch client and migrate all the APIs to that. We don't know when things might start to break. Because we know that normal search/index operations don't work with the ES client. But it looks like maintenance queries do work. So while things are okay for now, they might not be in the future.

Reply:
:point_up:

Reply:
This is safe for now given that we validate the `ElasticSearchSettings.Version` config to start with `Elasticsearch_`, which points to Elasticsearch-specific versions.

But yes, good point, when we support Opensearch we'll need to change this, so I'll create at ticket.

Reply:
https://mattermost.atlassian.net/browse/MM-57876

Reply:
But the actual implementation by AWS is still opensearch. So we are just using the ES client to point to an OS server.

Reply:
From what I've read, if you choose Elasticsearch with version <= 7.10, you get a 100% compatible implementation, but I may be mistaken, because this is *complex*.

Reply:
AWS doesn't use Elasticsearch anymore. It's all opensearch. Can you clarify from where you read that?

Reply:
Ahh you are right. Yeah so till 7.10 AWS allows real elasticsearch. But 7.x will eventually go EOL, at which point we'd need to use opensearch. 

@streamer45 - Apologies I was wrong in our 1:1 discussion. AWS allows actual elasticsearch, but only till 7.10 version. I will check with the relevant customer on this.